tests/python/pants_test/pantsd/test_pantsd_integration.py

```diff
@@ -168,14 +168,18 @@ def test_pantsd_run(self):
       pantsd_run(['list', '3rdparty:'])
       checker.await_pantsd()
 
+      # Some testprojects have warnings or errors, so skip them.
+      all_dirs_except_testprojects = ['3rdparty', 'contrib', 'src', 'tests', 'pants-plugins']
+      all_targets_except_testprojects = ['{}::'.format(d) for d in all_dirs_except_testprojects]
+
       pantsd_run(['list', ':'])
       checker.assert_running()
 
-      pantsd_run(['list', '::'])
+      pantsd_run(['list'] + all_targets_except_testprojects)
```

Comment:
this seems like an undesirable coverage regression from pantsd's perspective to me. could you not just plumb `build_file_imports_behavior='allow'` as the default for these tests, which shouldn't warn or throw errors?

Reply:
I can certainly do that. But other checks which we have in the repo explicitly exempt testprojects because we don't expect them to necessarily work or be valid. Given that, it seems that exempting them here is worthwhile, because otherwise next time someone adds an intentionally broken testproject, they'll need to touch this test again.

@kwlzn @stuhood WDYT?

Reply:
It's highly desirable to be able to write integration tests that are invalid. It's also highly desirable for running `./pants list ::` in the repo to not trigger warnings.

In general, to avoid having integration tests trigger warnings or failures, we've fiddled around with having non-BUILD files that are converted into BUILD files, and etc. That's slightly awkward, but it's also a potential solution here: https://github.com/pantsbuild/pants/blob/d30cca1e0ecb9cc0e1b7e2cd0ff6e7e077e62a52/tests/python/pants_test/backend/jvm/subsystems/test_jar_dependency_management_integration.py#L28-L31

Reply:
how about leaving the `list ::` but just checking for errors (`E`) vs warnings + errors (`WE`)?

the warnings check is just a historical remnant I believe, I'm not sure it buys us anything meaningful here today.

Reply:
Done